src/macros-by-example.adoc

```diff
@@ -631,24 +643,94 @@ just last place.
 [{mrk}]
 ----
 (*macro* prices
-  [(**number* **amount), (*symbol* currency)]
-  (*for_each* amount
-    (price amount currency)))
+  [(amount *number{asterisk}*), (currency *symbol!*)]
+  (*for* [(amt amount)]
+    (price amt currency)))
 ----
 
 The calling convention for `***` is different from `*\...*` since the “all the rest”
-convention can’t be used to draw the boundaries of the stream.  Instead, we use a list or
-S-expression as delimiting syntax to group the applicable sub-expressions:
+convention can’t be used to draw the boundaries of the stream.  Instead, we need a single
+expression that produces the desired values:
+
+[{mrk}]
+----
+(:prices (:) JPY)               ⇒ _void_
+(:prices 54 CAD)                ⇒ {amount:54, currency:CAD}
+(:prices (:values 10 9.99) GBP) ⇒ {amount:10, currency:GBP} {amount:9.99, currency:GBP}
+----
+
+
+==== One-or-More
+
+A parameter with the modifier `*+*` has _one-or-more cardinality_, which works like `***` except
+the resulting stream must produce at least one value.  To continue using our `prices` example:
+
+[{mrk}]
+----
+(*macro* prices
+  [(amount *number+*), (currency *symbol!*)]
+  (*for* [(amt amount)]
+    (price amt currency)))
+----
+
+[{mrk}]
+----
+(:prices (:) JPY) ⇒ _**error**: at least one value expected for + parameter_
+(:prices 54 CAD)                ⇒ {amount:54, currency:CAD}
+(:prices (:values 10 9.99) GBP) ⇒ {amount:10, currency:GBP} {amount:9.99, currency:GBP}
+----
+
+A macro's final parameter can use a variant of rest parameters with one-or-more cardinality,
+denoted by the `*\...+*` modifier:
+
+[{mrk}]
+----
+(*macro* thanks [(names *text\...+*)]
+  (make_string "Thank you to my Patreon supporters:\n"
+    (for [(n names)]
+      (make_string "  * " n "\n"))))
+----
 
 [{mrk}]
 ----
-(:prices () JPY)         ⇒ _void_
-(:prices (10  9.99) GBP) ⇒ {amount:10, currency:GBP} {amount:9.99, currency:GBP}
-(:prices [10, 9.99] GBP) ⇒ {amount:10, currency:GBP} {amount:9.99, currency:GBP}
+(:thanks) ⇒ _**error**: at least one value expected for \...+ parameter_
+
+(:thanks Larry Curly Moe) =>
+'''\
+Thank you to my Patreon supporters:
+  * Larry
+  * Curly
+  * Moe
+'''
+----
+
+
+=== Grouped Parameters
+
+The non-rest versions of multi-value parameters can be annoying to invoke, since they require the
+use of `values` or some other template to produce the stream of values.  To streamline invocation,
+a macro can opt-in to special syntax that uses a list as delimiting syntax to group the
+applicable sub-expressions.  This is denoted by wrapping the parameter's type in `[]`:
+
+[{mrk}]
+----
+(*macro* prices
+  [(amount *[number]{asterisk}*), (currency *symbol!*)]
```

Comment:
I was under the impression that `*` was the default for grouped expressions and the only explicit cardinality for those was `+`.

Reply:
Reading through the glossary changes in #217, I realize that I've been thinking this because `[]` is a grouping mode like `...`, and we don't require `...*`, only `...+`.

Reply:
I didn't remember what we resolved during the onsite, so I opted for consistency in always requiring that marker.  It did feel redundant, so I'd be happy to remove it.   Your point about `...` is a good supporting argument.

@almann Do you agree with making the `*` optional and default for `[]`?

Reply:
I am good with this.

Reply:
OK, I'll make that change in a separate CR.

Reply:
Updated in #218